compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/types/IrTypeUtils.kt

```diff
@@ -108,3 +108,19 @@ fun IrType.getArrayElementType(irBuiltIns: IrBuiltIns): IrType =
         irBuiltIns.primitiveArrayElementTypes[classifier]
             ?: throw AssertionError("Primitive array expected: $classifier")
     }
+
+fun IrType.toArrayType(irBuiltIns: IrBuiltIns): IrType =
+    if (isPrimitiveType()) {
+        irBuiltIns.primitiveArrayForType[this]?.defaultType
+            ?: throw AssertionError("$this not in primitiveArrayForType")
+    } else {
+        IrSimpleTypeBuilder().apply {
+            classifier = irBuiltIns.arrayClass
```

Comment:
This looses nullability information and I think that the return type of `arrayOf` is an invariant array. In this case, you could get away with the slightly shorter: `irBuiltIns.arrayClass.typeWith(this)`.

Reply:
Didn't know that simple way, awesome!